### PR TITLE
Karma: puppeeter support and native events

### PR DESCRIPTION
**Pull request by dvoytenko**
Reopened as https://github.com/google/web-stories-wp/pull/1812

Partial for #1721.

Blocked by #1695.

### CODE REVIEW

File: karma/karma-puppeteer-client/client.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * limitations under the License.
+ */
+
+;(function (global) {
+  /* global karmaPuppeteer */
+  'use strict'
+
+  function noCleanup() {}
+
+  function withSelector(methodName) {
```
Comment (cvolzke4) [inline karma/karma-puppeteer-client/client.js]: Can you add some comments for this function? It's not immediately obvious what this does.

Comment (dvoytenko) [reply karma/karma-puppeteer-client/client.js]: Will do. But for context: Puppeteer can only do APIs like this: `click('.selector')`. But I *also* want to enable here a simpler form: `click(element)`, b/c most of time you will already have an element available in the test. So, when an element is passed to these methods, I set a temporary unique attribute on it and instead pass the CSS selector to Puppeteer.

**Comment by dvoytenko** [comment]
@swissspidy @cvolzke4 this is ready for review. I recommend you review the #1695 first, however.

**Comment by dvoytenko** [comment]
Somehow I borked up the GitHub's base and it thinks the pull request is merged. Resending the same PR in the https://github.com/google/web-stories-wp/pull/1812

**Pull request merged**